<launch>
    <arg name="debug" default="false"/>

    <arg unless="$(arg debug)" name="launch-prefix" value=""/>
    <arg     if="$(arg debug)" name="launch-prefix" value="gdb ex --args"/>

    <env name="SMPL_VISUALIZE_CONFIG_FILE" value="$(find smpl_test)/config/smpl_visual.ini"/>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find smpl_test)/config/rosconsole.properties"/>

    <include file="$(find smpl_test)/launch/test_common.launch.xml"/>

    <node name="test_planner" pkg="smpl_test" type="test_planner"
            launch-prefix="$(arg launch-prefix)">
        <param name="robot_model_type" value="kdl"/>

        <rosparam param="graph" subst_value="true">
            type: manip_lattice
            discretization:
                r_shoulder_pan_joint    0.017453292519943295
                r_shoulder_lift_joint   0.017453292519943295
                r_upper_arm_roll_joint  0.017453292519943295
                r_elbow_flex_joint      0.017453292519943295
                r_forearm_roll_joint    0.017453292519943295
                r_wrist_flex_joint      0.017453292519943295
                r_wrist_roll_joint      0.017453292519943295
            use_long_dist_mprims: true
            use_xyz_snap_mprim: false
            use_rpy_snap_mprim: false
            use_xyzrpy_snap_mprim: true
            short_dist_mprims_thresh: 0.4
            xyz_snap_dist_thresh: 0.0
            rpy_snap_dist_thresh: 0.0
            xyzrpy_snap_dist_thresh: 0.1
            mprim_filename: $(find smpl_test)/config/pr2.mprim
        </rosparam>

        <param name="num_heuristics" value="2"/>

        <rosparam param="heuristic_0">
            type: bfs
            inflation_radius: 0.02
            cost_per_cell: 100
        </rosparam>

        <rosparam param="heuristic_1">
            type: bfs
            inflation_radius: 0.2
            cost_per_cell: 100
        </rosparam>

        <rosparam param="search">
            type: umhastar
            w_heur_init: 100.0
            anchor_expansion_freq: 10
        </rosparam>

        <rosparam param="goal">
            type: pose
            x: 0.4
            y: -0.2
            z: 0.36
            roll: 0.0
            pitch: 0.0
            yaw: 0.0
        </rosparam>

        <param name="allowed_planning_time" value="15.0"/>
    </node>
</launch>

